retail/interactive-tutorial/RetailEvents.Samples/PurgeUserEventSample.cs

```diff
@@ -0,0 +1,128 @@
+// Import user events into a catalog from inline source using Retail API
+
+using Google.Cloud.Retail.V2;
+using Google.LongRunning;
+using Google.Protobuf.WellKnownTypes;
+using System;
+
+/// <summary>
+/// The purge user event sample class.
+/// </summary>
+public class PurgeUserEventSample
+{
+    /// <summary>
+    /// The the user event.
+    /// </summary>
+    /// <returns>The usert event.</returns>defaultCatalog
+    public static UserEvent GetUserEvent()
+    {
+        ProductDetail productDetail = new ProductDetail
+        {
+            Product = new Product
+            {
+                Id = "test_id"
+            },
+            Quantity = 3
+        };
+
+        UserEvent userEvent = new UserEvent
+        {
+            EventType = "detail-page-view",
+            VisitorId = "test_visitor_id",
+            EventTime = new Timestamp
+            {
+                Seconds = DateTime.Now.ToUniversalTime().ToTimestamp().Seconds
+            }
+        };
+
+        userEvent.ProductDetails.Add(productDetail);
+
+        Console.WriteLine("User Event:");
+        Console.WriteLine($"Product detail: {userEvent.ProductDetails}");
+        Console.WriteLine($"Event type: {userEvent.EventType}");
+        Console.WriteLine($"Visitor id: {userEvent.VisitorId}");
+        Console.WriteLine($"Event time: {userEvent.EventTime}");
+        Console.WriteLine();
+
+        return userEvent;
+    }
+
+    /// <summary>
+    /// Get the purge user event request.
+    /// </summary>
+    /// <param name="defaultCatalog">The default catalog.</param>
+    /// <returns>The purge user event request.</returns>
+    private static PurgeUserEventsRequest GetPurgeUserEventsRequest(string defaultCatalog)
+    {
+        PurgeUserEventsRequest purgeRequest = new PurgeUserEventsRequest
+        {
+            Parent = defaultCatalog,
+            Filter = $"visitorId=\"test_visitor_id\"", // To check error handling set invalid filter here.
+            Force = true
+        };
+
+        Console.WriteLine("Purge user event. request:");
+        Console.WriteLine($"Parent: {purgeRequest.Parent}");
+        Console.WriteLine($"Filter: {purgeRequest.Filter}");
+        Console.WriteLine($"Force: {purgeRequest.Force}");
+        Console.WriteLine();
+
+        return purgeRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to purge user events.
+    /// </summary>
+    /// <param name="defaultCatalog">The default catalog.</param>
+    /// <returns>The completed purge response.</returns>
+    public static Operation<PurgeUserEventsResponse, PurgeMetadata> CallPurgeUserEvents(string defaultCatalog)
+    {
+        PurgeUserEventsRequest purgeRequest = GetPurgeUserEventsRequest(defaultCatalog);
+        UserEventServiceClient client = UserEventServiceClient.Create();
+        Operation<PurgeUserEventsResponse, PurgeMetadata> purgeResponse = client.PurgeUserEvents(purgeRequest);
+
+        Console.WriteLine("The purge operation was started:");
```

Comment:
_[this is duplicated reply from conversation above]_
@amanda-tarafa Initially, I was trying to use the same "PollingUntilComplete" as with Import samples, but after testing it, it was failing after many hours with this exception:
> Grpc.Core.RpcException : Status(StatusCode="Unknown", Detail="Stream removed", DebugException="Grpc.Core.Internal.CoreErrorDetailException: {"created":"@1647903062.960000000","description":"Error received from peer ipv4:172.217.18.106:443","file":"......\src\core\lib\surface\call.cc","file_line":1067,"grpc_message":"Stream removed","grpc_status":2}")

After analysis of the Purge/Rejoin API's and after short discussion with @tetiana-karasova I have found that those operations could take hours/days to complete, so, as I understand this is the main reason for such behavior, so we are just starting those operations and are not waiting for the completion.
What do you think? This is how it is also implemented in other languages.

Reply:
OK, let's do something like this then, so that the user can know what is happening, and same for the other operation

```suggestion
        Console.WriteLine("The purge operation was started.");
        
        // The purge operation takes several hours or even days to complete.
        // You may get the name of the operation
        string operationName = response.Name;
        // This name can be stored, then the long-running operation retrieved later by name
        Operation<PurgeUserEventsResponse, PurgeMetadata> retrievedResponse = client.PollOncePurgeUserEvents(operationName);
        // Check if the retrieved long-running operation has completed
        if (retrievedResponse.IsCompleted)
        {
            // If it has completed, then access the result
            PurgeUserEventsResponse retrievedResult = retrievedResponse.Result;
            Console.WriteLine(retrievedResult.Something);
        }
```